rllib/evaluation/rollout_worker.py

```diff
@@ -1580,13 +1591,21 @@ def set_weights(
             >>> # Set `global_vars` (timestep) as well.
             >>> worker.set_weights(weights, {"timestep": 42}) # doctest: +SKIP
         """
-        # If per-policy weights are object refs, `ray.get()` them first.
-        if weights and isinstance(next(iter(weights.values())), ObjectRef):
-            actual_weights = ray.get(list(weights.values()))
-            weights = {pid: actual_weights[i] for i, pid in enumerate(weights.keys())}
+        # Only update our weights, if no seq no given OR given seq no is different
+        # from ours
+        if weights_seq_no is None or weights_seq_no != self.weights_seq_no:
```

Comment:
Nit: If we have a sequence number, it would probably be expected behaviour to update if it increments.

Reply:
But that's what we do, no? Below in the line

```
self.weights_seq_no = weights_seq_no
```

We even update if the passed in seq_no is None, such that next time any seq-no is passed that's not None, we also update again and - after that - have a non-None seq-no again. :)